wal/wal.go

```diff
@@ -615,17 +619,21 @@ func (w *WAL) log(rec []byte, final bool) error {
 		copy(buf[recordHeaderSize:], part)
 		p.alloc += len(part) + recordHeaderSize
 
-		// By definition when a record is split it means its size is bigger than
-		// the page boundary so the current page would be full and needs to be flushed.
-		// On contrary if we wrote a full record, we can fit more records of the batch
-		// into the page before flushing it.
-		if final || typ != recFull || w.page.full() {
-			if err := w.flushPage(false); err != nil {
+		if w.page.full() {
+			if err := w.flushPage(true); err != nil {
 				return err
 			}
 		}
 		rec = rec[l:]
 	}
+
+	// If it's the final record of the batch and the page is not empty, flush it.
+	if final && !w.page.empty() {
```

Comment:
don't see a need for an extra func if it is called only in a single place.

Reply:
There are a few checks in this file like:
```
if w.page.alloc > 0 {
```
Should that logic all be inlined into `flushPage` such that it is a noop if there is no data to flush?

Reply:
Updated, if it is necessary, I'll inline it in another PR